Contribute/how-to-write-links.md

```diff
@@ -57,42 +77,269 @@ To create an inline link from a Docs technical article to another Docs technical
+> None of the previous examples use the `~/` as part of the link. To link to an absolute path that
+> begins at the root of the repository, start the link with `/`. Including the `~/` produces invalid
+> links when navigating the source repositories on GitHub. Starting the path with `/` resolves
+> correctly.
+
+### Structure of links on docs.microsoft.com
+
+Content published on docs.microsoft.com has the following URL structure:
+
+```
+https://docs.microsoft.com/<locale>/<product-service>/[<feature-service>]/[<subfolder>]/<topic>[?view=<view-name>]
+```
+
+Examples:
 
-To link to an article in a different docset, even if the file is in the same repository, use the following syntax:
+```
+https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-overview
+https://docs.microsoft.com/en-us/powershell/azure/overview?view=azurermps-5.1.1
+```
 
-`[link text](/docset-root/directory/article-name)`
-   
-For example, if an article whose root URL is `https://docs.microsoft.com/dotnet` links to an article whose root URL is `https://docs.microsoft.com/visualstudio`, the link would look like `[link text](/visualstudio/directory/article-name)`.
+- `<locale>` - identifies the language of the article (example: en-us or de-de)
+- `<product-service>` - the name of the product or service (example: powershell, dotnet, or azure)
+- `[<feature-service>]` - (optional) the name of the product's feature or subservice (example:
+  csharp or load-balancer)
+- `[<subfolder>]` - (optional) the name of a subfolder within a feature
+- `<topic>` - the name of the article file for the topic (example: load-balancer-overview or
+  overview)
+- `[?view=\<view-name>]` - (optional) the view name used by the version selector for content that
+  has multiple versions available (example: azps-3.5.0)
 
 > [!TIP]
-> Articles in the same *docset* have the same URL fragment after "docs.microsoft.com". For example, `https://docs.microsoft.com/dotnet/core/get-started` and `https://docs.microsoft.com/dotnet/framework/install` are in the same docset, and `https://docs.microsoft.com/dotnet/core/get-started` and `https://docs.microsoft.com/visualstudio/whats-new` are in different docsets.
 
-## Links to anchors
+> In most cases, articles in the same _docset_ have the same `<product-service>` URL fragment. For
+> example:
+> - Same docset
+>   - `https://docs.microsoft.com/dotnet/core/get-started`
+>   - `https://docs.microsoft.com/dotnet/framework/install`
+> - Different docset
+>   - `https://docs.microsoft.com/dotnet/core/get-started`
+>   - `https://docs.microsoft.com/visualstudio/whats-new`
+
+## Bookmark links
+
+For a bookmark link to a heading in the *current* file, use a hash symbol followed by the lowercase
```

Comment:
We should add a note/tip here about using the Docs Markdown extension to generate the bookmark link for you. https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-markdown

Reply:
I think we should linked to documentation on the feature in the Docs extension. But that feature is not documented anywhere. I added a note but we need a better link